yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1264,7 +1266,7 @@ object Client extends Logging {
 
       val secondaryJars =
         if (args != null) {
-          getSecondaryJarUris(Option(args.addJars).map(_.split(",").toSeq))
+          getSecondaryJarUris(sparkConf.get(JARS_TO_DISTRIBUTE).map(_.split(",").toSeq))
```

Comment:
From my understanding, the jar paths of `--jars` already resolved in `SparkSubmitArguments`, so here I'm not sure why and how we keep the pre-processed values.